### PR TITLE
Use common functions for interacting with shim

### PR DESCRIPTION
**Pull request by GarySmith**
Use a common set of functions for making REST fetch, post, and delete
calls to the shim.  Several advantages of this approach are:
- handle a known race condition where the UI needs to retrieve the URL
  to the shim (in an asyncronous function that fetches the config file)
  and to use the shim URL (in any asyncronous function that interacts
  with the shim).  The common functions use a chain of promises to
  eliminate this race condition.
- unify REST response handling, including extraction and conversion of
  results and error responses
- simplify and unify REST request headers so that a common, well-known
  set are used

### CODE REVIEW

File: src/pages/AssignServerRoles.js [hunk 1]
```diff
@@ -652,58 +627,16 @@ class AssignServerRoles extends BaseWizardPage {
...
-        headers: {
-          'Accept': 'application/json, text/plain, */*',
-          'Content-Type': 'application/json'
-        },
-        body: JSON.stringify(server)
-      })
-        .then((response) => this.checkResponse(response))
-    );
-  }
-
```
Comment (guglo) [inline src/pages/AssignServerRoles.js]: it still call this updateDiscoveredServer at line 938

Comment (GarySmith) [reply src/pages/AssignServerRoles.js]: Thanks for catching that.  Will address it.

File: src/pages/ValidateConfigFiles.js [hunk 2]
```diff
@@ -185,44 +176,18 @@ class ValidateConfigFiles extends Component {
...
-          } else {
-            this.setState({invalidMsg: responseData.log});
-          }
-        }
+      .catch(resp => {
+        this.props.enableNextButton(false);
+        resp.json().then(reason =>
+          this.setState({valid: INVALID, invalidMsg: reason.log}));
       });
   }
```
Comment (guglo) [inline src/pages/ValidateConfigFiles.js]: I got some errors when I run validate error from backend
 
INFO:ardana_service:2017-11-28 18:56:04.223087 127.0.0.1     POST http://localhost:8081/api/v2/config_processor
ERROR:ardana_service.config_processor:Command '['/home/gugl/dev/day0React/ardana-service/tools/mock-cmd', '/home/gugl/dev/day0React/ardana-service/data/config-processor/bin/python', '/home/gugl/dev/day0React/ardana-service/data/config-processor/ardana-configuration-processor/Driver/ardana-cp', '-l', '/tmp/tmpFWKCXd/log', '-c', '/home/gugl/dev/day0React/ardana-service/data/my_cloud/model/cloudConfig.yml', '-s', '/home/gugl/dev/day0React/ardana-service/data/ardana-input-model/2.0/services', '-r', '/home/gugl/dev/day0React/ardana-service/data/config-processor/ardana-configuration-processor/Data/Site', '-o', '/tmp/tmpFWKCXd/clouds']' returned non-zero exit status 254
Traceback (most recent call last):
  File "/home/gugl/dev/day0React/ardana-service/ardana_service/config_processor.py", line 93, in run_config_processor
    subprocess.check_output(cmd, stderr=subprocess.STDOUT)
  File "/usr/lib64/python2.7/subprocess.py", line 219, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
CalledProcessError: Command '['/home/gugl/dev/day0React/ardana-service/tools/mock-cmd', '/home/gugl/dev/day0React/ardana-service/data/config-processor/bin/python', '/home/gugl/dev/day0React/ardana-service/data/config-processor/ardana-configuration-processor/Driver/ardana-cp', '-l', '/tmp/tmpFWKCXd/log', '-c', '/home/gugl/dev/day0React/ardana-service/data/my_cloud/model/cloudConfig.yml', '-s', '/home/gugl/dev/day0React/ardana-service/data/ardana-input-model/2.0/services', '-r', '/home/gugl/dev/day0React/ardana-service/data/config-processor/ardana-configuration-processor/Data/Site', '-o', '/tmp/tmpFWKCXd/clouds']' returned non-zero exit status 254
INFO:ardana_service:2017-11-28 18:56:08.432145 127.0.0.1 400 POST http://localhost:8081/api/v2/config_processor

error from UI

POST http://localhost:8081/api/v1/clm/config_processor 400 (BAD REQUEST)
index.js:115197 Uncaught (in promise) TypeError: resp.json is not a function
    at index.js:115197
    at <anonymous>

Comment (guglo) [reply src/pages/ValidateConfigFiles.js]: I re-selected a different model...the error is gone. Not sure what exactly caused the previous error, maybe just a fluke

**Comment by guglo** [comment]
I tested a couple of pages..mostly it seems to function as normal except the updateDiscoveredServer. Also it needs to rebase as well.

**Comment by guglo** [comment]
@GarySmith  I don't not even know how to set request change, I clicked request change...but it doesn't show up.  Anyway,  think you know.

**Comment by GarySmith** [comment]
Rebased and addressed the problem updating discovered servers

**Comment by GarySmith** [comment]
Gloria was able to reproduce the bug above by running the config processor on an invalid model, causing some broken error handling logic to be executed. Will address this and update this change request.

**Pull request merged**